internal/pkg/cli/pipeline_init.go

```diff
@@ -545,13 +562,13 @@ func (o *initPipelineOpts) createPipelineManifest() error {
 		stages = append(stages, stage)
 	}
 
-	manifest, err := manifest.NewPipeline(pipelineName, provider, stages)
+	manifest, err := manifest.NewPipeline(o.name, provider, stages)
 	if err != nil {
 		return fmt.Errorf("generate a pipeline manifest: %w", err)
 	}
 
 	var manifestExists bool
-	manifestPath, err := o.workspace.WritePipelineManifest(manifest)
+	manifestPath, err := o.workspace.WritePipelineManifest(manifest, o.name)
```

Comment:
Question: do we need to pass in the name here? It should already be in the manifest after `manifest.NewPipeline`, right? Or is it for the `PipelineManifest` object?

Reply:
Yeah, we need it - [`WritePipelineManifest` takes a `encoding.BinaryMarshaler`](https://github.com/aws/copilot-cli/commit/f84e064aff6dd1fe0d1d82bb79c2dcfcb296bc39#diff-28e31e11ec2b0ddd8baf047d045914d541968c933e5bad45a2afed259be99551), so it doesn't know that `manifest` has a `Name` field. 

So we could change `wsPipelineWriter` to accept a `manifest.Pipeline` instead of the `BinaryMarshaler` to get rid of the name argument, which I would be fine doing. However, the [Service/Job manifest writing uses the same logic](https://github.com/aws/copilot-cli/blob/mainline/internal/pkg/workspace/workspace.go#L313-L328) as what's here currently. 

Thoughts?

Reply:
Ah, gotcha gotcha. 
I think it's fine to leave as is. 
This is only tangentially related and more a note to myself than anything, but let's remember to consider, as we progress into next steps, the potential mismatch of path pipeline name and manifest pipeline name if a user changes it between `init` and `deploy`.